File: euclidean_gcd/euclidean_gcd.js
function euclideanGcdRecursive(first, second) {
	if(second === 0) {
		return first;
	} else {
		return euclideanGcdRecursive(second, (first % second));
	}
}

function euclideanGcd(first, second) {
	while(second !== 0) {
		let temp = second;
		second = first % second;
		first = temp;
	}
	return first;
}

function main() {
	let first = 20;
	let second = 30;
	console.log('Recursive GCD for %d and %d is %d', first, second ,euclideanGcdRecursive(first, second));
	console.log('Iterative GCD for %d and %d is %d', first, second, euclideanGcd(first, second));
}

main();

File: euclidean_gcd/euclidean_gcd.go
package main

import "fmt"

// EuclideanGCD finds GCD of given numbers by iterative method
// first : first number
// second : second number
// Time complexity :O(log min(first,second))
func EuclideanGCD(first int, second int) int {
	for second != 0 {
		tmp := second
		second = first % second
		first = tmp
	}
	return first
}

// EuclideanGCDRecursive finds GCD of given numbers by recursion
// first : first number
// second : second number
// Time complexity : O(log min(first,second))
func EuclideanGCDRecursive(first int, second int) int {
	if second != 0 {
		return EuclideanGCDRecursive(second, (first % second))
	}
	return first
}

func main() {
	first := 90
	second := 65
	fmt.Println("GCD of numbers is :")
	fmt.Println(EuclideanGCD(first, second))
	fmt.Println(EuclideanGCDRecursive(first, second))
}
